lib/dynamoid/persistence.rb

```diff
@@ -145,6 +147,22 @@ def dynamo_type(type)
         end
       end
 
+      private
+
+      def transform_hash(hash)
+        {}.tap do |h|
+          hash.each { |key, value| h[key.to_sym] = transform(value) }
+        end
+      end
+
+      def transform(val)
+        case val
+        when BigDecimal; val.to_f
```

Comment:
This might need to be a configuration option. I could imagine other users of the world not wanting to always cast to a float.